tests/widget/modal/widget-modal.login.test.js

```diff
@@ -22,3 +22,54 @@ test('modal widget', async ({ page }) => {
   expect(await fullName.innerText()).toBe('Full name: Demo User');
   expect(await email.innerText()).toBe('Email: demo@user.com');
 });
+
+test.only('modal widget accessibly', async ({ page }) => {
+  const loginButton = page.locator('button', { hasText: 'Open Login Modal' });
+  const dialog = page.locator('dialog');
+  expect(await dialog.isVisible()).toBeFalsy();
+
+  await loginButton.click();
+  // Add just a bit of a delay to ensure dialog responds
+  await page.waitForTimeout(1000);
+  expect(await dialog.isVisible()).toBeTruthy();
+  await page.goto('widget/modal');
+  const username = page.getByRole('textbox', { name: /User Name/ });
+  const password = page.getByRole('textbox', { name: /Password/ });
+  const eye = page.getByRole('button', { name: 'show-password' });
+
+  const loginBtn = page.getByRole('button', { name: 'Sign in' });
+  await page.keyboard.press('Tab')
+  expect(username).toBeFocused();
+  await page.type(username, 'username01');
+
+  await page.keyboard.press('Tab')
+  expect(password).toBeFocused();
+  await page.type(password, 'password');
```

Comment:
Can we split the username and password from each other (i.e. don't use a page node)? I'd like this a11y test to encounter a multi-step journey to test our focus management.